### PR TITLE
CDAP-7486 Use TMS instead of Kafka for Tracker

### PR DESCRIPTION
**Pull request by gokulavasan**
JIRA : https://issues.cask.co/browse/CDAP-7486

Build : http://builds.cask.co/browse/TRAC-BT28-1

Note that the build won't pass until the latest CDAP release artifacts are available. I confirmed that the build passes locally (with locally installed CDAP artifacts)

### CODE REVIEW

**Comment by gokulavasan** [comment]
Investigating some test failures.

File: pom.xml [hunk 1]
```diff
@@ -178,14 +178,20 @@
...
-      <artifactId>cdap-kafka-flow-compat-0.8</artifactId>
-      <version>${kafka.pack.version}</version>
-    </dependency>
-    <dependency>
-      <groupId>co.cask.cdap</groupId>
       <artifactId>cdap-proto</artifactId>
       <version>${cdap.version}</version>
     </dependency>
+    <dependency>
+      <groupId>org.apache.twill</groupId>
```
Comment (anwar6953) [inline pom.xml]: why does it need this dependency?
Shouldn't cdap-api be sufficient?

Comment (gokulavasan) [inline pom.xml]: DiscoveryMetadataClient and some other stuff in TrackerApp needs twill zookeeper classes.

**Comment by chtyim** [comment]
LGTM

**Comment by gokulavasan** [comment]
Thanks for the review @chtyim. Can you take a look at the final commit (https://github.com/caskdata/cask-tracker/pull/82/commits/a617b681c1fb3e4a8594fbaab96f331235c0ca9a) which contains some optimizations on fetching messages from TMS.

**Comment by chtyim** [comment]
LGTM for the last commit.

**Pull request merged**